tests/test_networkx/conftest.py

```diff
@@ -0,0 +1,123 @@
+from pathlib import Path
+from typing import Any, Literal, TypedDict
+
+import networkx as nx
+import numpy as np
+import pytest
+from numpy.typing import NDArray
+
+import geff
+
+DTypeStr = Literal["double", "int", "int8", "uint8", "int16", "uint16", "float32", "float64"]
+Axes = Literal["t", "z", "y", "x"]
+
+
+class GraphAttrs(TypedDict):
+    nodes: NDArray[Any]
+    edges: NDArray[Any]
+    node_positions: NDArray[Any]
+    extra_node_attrs: dict[str, NDArray[Any]]
+    edge_attrs: dict[str, NDArray[Any]]
+    directed: bool
+    axis_names: tuple[Axes, ...]
+    axis_units: tuple[str, ...]
+
+
+class ExampleNodeAttrs(TypedDict):
+    position: DTypeStr
+
+
+class ExampleEdgeAttrs(TypedDict):
+    score: DTypeStr
+    color: DTypeStr
+
+
+def create_dummy_graph_attrs(
+    node_dtype: DTypeStr,
+    node_attr_dtypes: ExampleNodeAttrs,
+    edge_attr_dtypes: ExampleEdgeAttrs,
+    directed: bool,
+) -> GraphAttrs:
+    axis_names = ("t", "z", "y", "x")
+    axis_units = ("s", "nm", "nm", "nm")
+    nodes = np.array([10, 2, 127, 4, 5], dtype=node_dtype)
+    positions = np.array(
+        [
+            [0.1, 0.5, 100.0, 1.0],
+            [0.2, 0.4, 200.0, 0.1],
+            [0.3, 0.3, 300.0, 0.1],
+            [0.4, 0.2, 400.0, 0.1],
+            [0.5, 0.1, 500.0, 0.1],
+        ],
+        dtype=node_attr_dtypes["position"],
+    )
+
+    edges = np.array(
+        [
+            [10, 2],
+            [2, 127],
+            [2, 4],
+            [4, 5],
+        ],
+        dtype=node_dtype,
+    )
+    scores = np.array([0.1, 0.2, 0.3, 0.4], dtype=edge_attr_dtypes["score"])
+    colors = np.array([1, 2, 3, 4], dtype=edge_attr_dtypes["color"])
+
+    return {
+        "nodes": nodes,
+        "edges": edges,
+        "node_positions": positions,
+        "extra_node_attrs": {},
+        "edge_attrs": {"score": scores, "color": colors},
+        "directed": directed,
+        "axis_names": axis_names,
+        "axis_units": axis_units,
+    }
+
+
+@pytest.fixture()
+def path_w_expected_graph_attrs(tmp_path) -> tuple[Path, GraphAttrs]:
```

Comment:
I'm wondering if it would be better to just have this as a normal function so that we can take advantage of the parametrization of different dtypes?